src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/EqualsAvoidNullCheck.java

```diff
@@ -154,18 +163,23 @@ public void visitToken(final DetailAST ast) {
     @Override
     public void leaveToken(DetailAST ast) {
         final int astType = ast.getType();
-        if (astType != TokenTypes.VARIABLE_DEF
+        if (astType == TokenTypes.SLIST) {
+            leaveSlist(ast);
+        }
+        else if (astType == TokenTypes.LITERAL_NEW) {
+            leaveLiteralNew(ast);
+        }
+        else if (astType == TokenTypes.OBJBLOCK) {
+            final int parentType = ast.getParent().getType();
+            if (parentType != TokenTypes.CLASS_DEF && parentType != TokenTypes.ENUM_DEF) {
```

Comment:
This is needed because the current code needs the class fields and methods on CLASS_DEF frame and not the OBJBLOCK frame.